root/io/json/runMap.C

```diff
@@ -0,0 +1,46 @@
+#include "TBufferJSON.h"
+#include <map>
+#include <string>
+
+void testJson(std::map<std::string,int> &data, int compact, const char *name)
+{
+   auto json = TBufferJSON::ToJSON(&data, compact);
+
+   std::cout << name << json << std::endl;
+
+   auto mread = TBufferJSON::FromJSON<std::map<std::string,int>>(json.Data());
+   if (!mread) {
+      std::cout << "Fail to read map from JSON" << std::endl;
+   } else if (data.size() != mread->size()) {
+      std::cout << "Mismatch in maps size " << data.size() << "  " << mread->size() << std::endl;
+   } else {
+      bool isok = true;
+      for (auto &item : data) {
+         if (mread->at(item.first) != item.second) {
+            std::cout << "Failure with field " << item.first << std::endl;
+            isok = false;
+         }
+      }
+      if (isok)
+         std::cout << "Data matches" << std::endl;
+   }
+
+}
+
+void runMap()
+{
+   std::map<std::string,int> data;
+
+   for (int n=0;n<10;++n) {
+      std::string key = "field";
+      key.append(std::to_string(n));
+      data[key] = n*7;
+   }
+
+   testJson(data, 0, "DEFAULT: ");
+
+   testJson(data, 5, "OBJECT: ");
```

Comment:
My apologies, I missed this details in the original PR.   Could we add an enum type for the user to be able to use semantic rather than magic number ?

Reply:
These "magic" numbers are since the beginning of TBufferJSON.
I will add them in separate PR

Reply:
thanks.

Reply:
I modify test to use introduced constants

Reply:
thanks.